### DISCUSSION TITLE
how to use this framework as a silent background process ?

### DISCUSSION THREAD

**Discussion by nukadelic** [cmbruns/pyopenxr#49] [General]
I want to record the hmd and the controllers positions and rotations into a csv file while playing a steam-vr game , would it be possible at the current state of the library ? where should i start looking how to do that ?

**Comment by cmbruns** [comment]
If you can figure out how to do that using the OpenXR SDK in C, we can probably figure out how to make it work in pyopenxr. Otherwise you might need to wait a long time.

**Comment by cmbruns** [comment]
I moved this to the discussions section, since it's more of a question than an issue

**Comment by jherico** [comment]
The API isn't really set up to do this kind of thing.  Most of the information you're looking for requires you to call [xrSyncActions](https://www.khronos.org/registry/OpenXR/specs/1.0/man/html/xrSyncActions.html), which in turn requires not only a session, but one which is in the FOCUSED state, meaning it currently presenting frames to the headset.

If what you want is to capture information while _other_ VR apps are active, your best bet is probably to write a layer.  Applications can request layers, but you can also activate a layer systemwide through other means (usually a registry or environment variable).  Your layer can then capture calls and pass information to whatever persistence setup you want.  But you're not going to be able to write an XR client application that captures pose information while other apps are active and presenting to the HMD.
